module/documents/actor/actor.mjs

```diff
@@ -1168,22 +1169,28 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
     }
 
     const relevant = type === "skill" ? this.system.skills?.[config.skill] : this.system.tools?.[config.tool];
+    const abilityId = relevant?.ability ?? (type === "skill" ? skillConfig.ability : toolConfig.ability);
+    const ability = this.system.abilities?.[abilityId];
     const hostActor = this.isPolymorphed && this.flags?.dnd5e?.transformOptions?.mergeSkills && (type === "skill")
       ? game.actors.get(this.flags.dnd5e?.originalActor) : null;
     const buildConfig = this._buildSkillToolConfig.bind(this, type, hostActor);
 
+    const { advantage, disadvantage } = AdvantageModeField.combineFields(this.system, [
```

Comment:
Do we need to combine modes here? We will ultimately invoke `#rollD20Test` which will check the advantage mode on the ability. Is it enough to get `advantage`, `disadvantage`, `minimum`, and `maximum` from the skill/tool here, then forward it to `#rollD20Test` which will check against the ability and combine it with the `advantage`, `disadvantage`, `minimum`, and `maximum` that it gets passed?

Same goes for concentration.